Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -800,25 +826,39 @@ protected bool CheckForCollisionBetween()
                 Map.debugDrawer.FlashLine(lastPosIV3, newPosIV3);
             }
             #endregion
-
+            var interceptor = BlockerRegistry.CheckForCollisionBetweenCallback(this, PreLastPos, ExactPosition);
+            if (interceptor.HasValue)
+            {
+                possibleIntersections.Add(interceptor.Value);
+            }
+            if (possibleIntersections.Any())
+            {
+                newPosIV3 = possibleIntersections.OrderBy(x => (LastPos - x.IntersectionPos).MagnitudeHorizontalSquared()).First().IntersectionPos.ToIntVec3();
+            }
             // Iterate through all cells between the last and the new position
             // INCLUDING[!!!] THE LAST AND NEW POSITIONS!
             var cells = GenSight.PointsOnLineOfSight(lastPosIV3, newPosIV3).Union(new[] { lastPosIV3, newPosIV3 }).Distinct().OrderBy(x => (x.ToVector3Shifted() - LastPos).MagnitudeHorizontalSquared());
-
             //Order cells by distance from the last position
             foreach (var cell in cells)
             {
-                if (CheckCellForCollision(cell))
-                {
-                    newPosIV3 = cell;
-                    collided = true;
-                    break;
-                }
+                var target = PossibleTargetForCell(cell);
 
                 if (Controller.settings.DebugDrawInterceptChecks)
                 {
                     Map.debugDrawer.FlashCell(cell, 1, "o");
                 }
+                if (target.HasValue)//if we found any target close, we don't need to check other
+                {
+                    possibleIntersections.Add(target.Value);
```

Comment:
This overcomplicates the API for per-cell checking considerably.  We already sorted the list of possible interceptors and selected the closest as the *last cell* to check.  If any cell before that collides, it will *always* be the first and have its action selected.  

If there is some strong reason to complicate this inner bit, it at the least should not be put onto the *end* of the list and then sorted to the front with a subsequent sort call.  Check if it is closer than the now front of the list, and replace.  But even there, we only need the first interceptor so there's no reason for the list at all.